faucet/valve.py

```diff
@@ -1361,6 +1366,23 @@ def get_config_dict(self):
             'acls': acls_dict,
             }
 
+    def flow_timeout(self, table_id, match):
+        eth_src = None
+        match_oxm_fields = match.to_jsondict()['OFPMatch']['oxm_fields']
+        if table_id == self.dp.eth_src_table:
```

Comment:
What if the flow times out in the eth_dst table? Could you clarify for me how you are avoiding the rule in eth_src but not eth_dst table situation. 

Cause it's nice to reduce packet_ins but we **MUST NOT** allow ever having addresses time out in the eth_dst table before the eth_src. Unfortunately it is very hard for the test suite to cover that situation, if you can think of a way to test it I would appreciate that.